sematic/resolvers/resource_requirements.py

```diff
@@ -286,9 +286,26 @@ class KubernetesResourceRequirements:
     mount_expanded_shared_memory: bool = field(default=False)
     security_context: Optional[KubernetesSecurityContext] = None
 
+    def clone(self) -> "KubernetesResourceRequirements":
+        """Deep copy these requirements"""
+        # only need to replace things that are mutable.
+        return replace(
+            self,
+            node_selector=dict(self.node_selector),
+            requests=dict(self.requests),
+            tolerations=[t for t in self.tolerations],
+        )
+
 
 @dataclass
 class ResourceRequirements:
     kubernetes: KubernetesResourceRequirements = field(
         default_factory=KubernetesResourceRequirements
     )
+
+    def clone(self) -> "ResourceRequirements":
```

Comment:
This and other changes in this file are to enable better deep cloning of futures.